tests/ops/pulse/test_convenience_functions.py

```diff
@@ -0,0 +1,432 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Unit tests for the convenience functions used in pulsed programming.
+"""
+import inspect
+from functools import reduce
+
+import pytest
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.ops.op_math import ParametrizedHamiltonian
+
+pytest.importorskip("jax")
```

Comment:
There have been some complaints about how tests run locally fail now that we have added the import tests for `jax`, since the local test environment has `jax` installed. 

This is how the same thing is dealt with in the test file for `mpldrawer`, which has a similar dependency (just on `matplotlib`). I don't know if its a good solution, but that's why I removed the jax import tests and added this instead. Thoughts?

Reply:
When running tests locally, one can do:
```
>>> pytest tests -m core
```
to skip all interface-related tests.

Reply:
If you have a good solution to test the `raise ValueError` when jax is not installed using the same file, then I have nothing against this.

But I'd like to avoid adding extra files for just one test.

Reply:
I'll remove and put it back to testing the import error here

Reply:
How about using `@pytest.mark.skipif("jax" in sys.modules.keys(), reason="fails if jax is installed")` to skip when running locally?